iree/modules/strings/api.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IREE_MODULES_STRINGS_STRINGS_API_H_
+#define IREE_MODULES_STRINGS_STRINGS_API_H_
+
+#include "iree/base/api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+typedef struct string {
+  iree_vm_ref_object_t ref_object;
+  iree_allocator_t allocator;
+  iree_string_view_t value;
+} string_t;
+
+typedef struct string_tensor {
```

Comment:
Do these need to be public in the header?

Reply:
I moved to api_detail.h to try to hide it away (similar to what we do for the hal).